### PR TITLE
Add an environment class with expressive typing.

### PR DESCRIPTION
**Pull request by joeleu**
This PR adds a class and default for presenting our environment in a more expressive way. It provides easy to use booleans where we can branch specific executions (e.g. the static and media settings). Ideally this will make our various environments easier to use.

### CODE REVIEW

**Comment by joeleu** [comment]
*sigh* I'll address the tests tomorrow.

**Review by Hgersten-hash — COMMENTED** [review]
This is a smart way to enable three different environment types. As was indicated by the test failure, we need to make it work with sending emails. I think this is because, in your code, the AWS_SES_DOMAIN isn't defined when the environment is local. Maybe, we can have it defined regardless of the environment type, and then check if it's empty in the domain part of the emails?

File: voluncheer/voluncheer/settings.py [hunk 1]
```diff
@@ -153,34 +155,34 @@
...
+if not environment.is_local:
+    EMAIL_BACKEND = "django_ses.SESBackend"
+    AWS_ACCESS_KEY_ID = os.getenv("AWS_SES_ACCESS_KEY")
+    AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SES_SECRET_ACCESS_KEY")
+    AWS_SES_REGION_NAME = os.getenv("AWS_SES_REGION_NAME")
+    AWS_SES_REGION_ENDPOINT = os.getenv("AWS_SES_REGION_ENDPOINT")
+    AWS_SES_DOMAIN = os.getenv("AWS_SES_DOMAIN")
+    DEFAULT_FROM_EMAIL = os.getenv("AWS_SES_FROM_EMAIL")
+else:
+    EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
```
Comment (JinweiZhu) [inline voluncheer/voluncheer/settings.py:168]: Since the test is failing for import, we can just use placeholder values like "N/A".

Comment (joeleu) [reply voluncheer/voluncheer/settings.py:166]: An empty string is as useful as nonsense so I left it empty. We can't really test email locally right now anyway.

**Pull request merged**